plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs

```diff
@@ -284,60 +385,28 @@ type HasConstant term = (AsConstant term, FromConstant term)
 -- and connects @term@ and its @uni@.
 type HasConstantIn uni term = (UniOf term ~ uni, HasConstant term)
 
--- | Extract the 'Constant' from a 'Term'
--- (or throw an error if the term is not a 'Constant' or the constant is not of the expected type).
-unliftConstant
-    :: forall a m term err.
-       (MonadError (ErrorWithCause err term) m, AsUnliftingError err, KnownBuiltinType term a)
-    => term -> m a
-unliftConstant term = case asConstant term of
-    Just (Some (ValueOf uniAct x)) -> do
-        let uniExp = knownUni @(UniOf term) @a
-        case uniAct `geq` uniExp of
-            Just Refl -> pure x
-            Nothing   -> do
-                let err = fromString $ concat
-                        [ "Type mismatch: "
-                        , "expected: " ++ gshow uniExp
-                        , "; actual: " ++ gshow uniAct
-                        ]
-                throwingWithCause _UnliftingError err $ Just term
-    Nothing -> throwingWithCause _UnliftingError "Not a constant" $ Just term
-
-{- Note [KnownType's defaults]
-We'd like to use @default@ for providing instances for built-in types instead of @DerivingVia@,
-because the latter breaks on @m a@
-
-It's possible to circumvent this, by using @forall r. (a -> m r) -> m r@ instead,
-but then we need to recover the original handy definitions and make the API and
-the code bloated (instances are more verbose with @DerivingVia@).
-
-We don't use @default@ in 'KnownTypeAst', because an attempt to write
-
-    default toTypeAst :: (k ~ GHC.Type, uni `Includes` a) => proxy a -> Type TyName uni ()
-    toTypeAst = toBuiltinTypeAst
-
-results in
-
-    Expected kind '* -> *', but 'uni' has kind 'k -> *'
-
-see https://gitlab.haskell.org/ghc/ghc/-/issues/15710
```

Comment:
We no longer have this issue, since `Contains` (previously named `Includes`) is kind-polymorphic and so we don't need to add a type equality constraint to the default implementation of `toTypeAst` and it just works.